resources/aws.go

```diff
@@ -0,0 +1,334 @@
+					svc.WaitUntilInstanceTerminated(&waitInput)
+				}
+			}
+
+		}
+	}()
+	for {
+		select {
+		case event := <-obj.Events():
+			if exit, send = obj.ReadEvent(event); exit != nil {
+				return *exit
+			}
+		case state := <-awschan:
+			log.Printf(state)
+			obj.StateOK(false)
+			send = true
+			if send {
+				send = false
+				obj.Event()
+			}
+		}
+	}
+}
+
+// CheckApply method for Aws resource.
+func (obj *AwsRes) CheckApply(apply bool) (checkOK bool, err error) {
+	log.Printf("%s: CheckApply(%t)", obj, apply)
+
+	sess := session.Must(session.NewSession(&aws.Config{
+		Region: aws.String(obj.Region),
+	}))
+	svc := ec2.New(sess)
+
+	diInput := ec2.DescribeInstancesInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("tag:Name"),
+				Values: []*string{aws.String(obj.GetName())},
+			},
+			{
+				Name:   aws.String("instance-state-name"),
+				Values: []*string{aws.String("running"), aws.String("pending"), aws.String("stopped"), aws.String("stopping")},
+			},
+		},
+	}
+	diOutput, err := svc.DescribeInstances(&diInput)
+	if err != nil {
+		return false, errwrap.Wrapf(err, "error describing instances")
+	}
+
+	if len(diOutput.Reservations) < 1 && obj.State == "terminated" {
+		return true, nil
+	}
+	if len(diOutput.Reservations) == 1 && *diOutput.Reservations[0].Instances[0].State.Name == obj.State {
+		return true, nil
+	}
+	if !apply {
+		return false, nil
+	}
+
+	if len(diOutput.Reservations) > 1 {
+		return false, fmt.Errorf("too many reservations")
+	}
+	if len(diOutput.Reservations) == 1 {
+		instanceID := diOutput.Reservations[0].Instances[0].InstanceId
+		describeInput := ec2.DescribeInstancesInput{
+			InstanceIds: []*string{instanceID},
+		}
+		if len(diOutput.Reservations[0].Instances) > 1 {
+			return false, fmt.Errorf("more than one instance was returned")
+		}
+		if obj.State == "running" {
+			startInput := ec2.StartInstancesInput{
+				InstanceIds: []*string{instanceID},
+			}
+			_, err := svc.StartInstances(&startInput)
+			if err != nil {
+				return false, errwrap.Wrapf(err, "error starting instance")
+			}
+			svc.WaitUntilInstanceRunning(&describeInput)
+			log.Println("instance running")
+		}
+		if obj.State == "stopped" {
+			stopInput := ec2.StopInstancesInput{
+				InstanceIds: []*string{instanceID},
+			}
+			_, err := svc.StopInstances(&stopInput)
+			if err != nil {
+				return false, errwrap.Wrapf(err, "error stopping instance")
+			}
+			svc.WaitUntilInstanceStopped(&describeInput)
+			log.Println("instance stopped")
+		}
+		if obj.State == "terminated" {
+			terminateInput := ec2.TerminateInstancesInput{
+				InstanceIds: []*string{instanceID},
+			}
+			_, err := svc.TerminateInstances(&terminateInput)
+			if err != nil {
+				return false, errwrap.Wrapf(err, "error terminating instance")
+			}
+			svc.WaitUntilInstanceTerminated(&describeInput)
+			log.Println("instance terminated")
+		}
+	}
+	if len(diOutput.Reservations) < 1 && obj.State == "running" {
+		minCount := int64(1)
```

Comment:
These are a bit awkward looks like you can instead use:

```
func (s *RunInstancesInput) SetMinCount(v int64) *RunInstancesInput
```

and chain these in...

Reply:
done